src/pdf/intent.rs

```diff
@@ -0,0 +1,90 @@
+use mupdf_sys::*;
+
+pub enum Intent {
+    Default,
+    FreetextCallout,
+    FreetextTypewriter,
+    LineArrow,
+    LineDimension,
+    PolylineDimension,
+    PolygonCloud,
+    PolygonDimension,
+    StampImage,
+    StampSnapshot,
+}
+
+impl From<Intent> for pdf_intent {
+    fn from(value: Intent) -> Self {
+        match value {
+            Intent::Default => PDF_ANNOT_IT_DEFAULT,
+            Intent::FreetextCallout => PDF_ANNOT_IT_FREETEXT_CALLOUT,
+            Intent::FreetextTypewriter => PDF_ANNOT_IT_FREETEXT_TYPEWRITER,
+            Intent::LineArrow => PDF_ANNOT_IT_LINE_ARROW,
+            Intent::LineDimension => PDF_ANNOT_IT_LINE_DIMENSION,
+            Intent::PolylineDimension => PDF_ANNOT_IT_POLYLINE_DIMENSION,
+            Intent::PolygonCloud => PDF_ANNOT_IT_POLYGON_CLOUD,
+            Intent::PolygonDimension => PDF_ANNOT_IT_POLYGON_DIMENSION,
+            Intent::StampImage => PDF_ANNOT_IT_STAMP_IMAGE,
+            Intent::StampSnapshot => PDF_ANNOT_IT_STAMP_SNAPSHOT
+        }
+    }
+}
+
+#[derive(Debug)]
+pub struct OutOfRange;
+
+impl std::fmt::Display for OutOfRange {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "The provided value was out of the allowed range")
+    }
+}
+
+impl std::error::Error for OutOfRange {}
+
+impl TryFrom<pdf_intent> for Intent {
```

Comment:
Couldn't this file be simplified by using the `from_enum!` macro like [here](https://github.com/messense/mupdf-rs/blob/5fc9003f9be480caee71d183df8fc3e9b7b74ae3/src/text.rs#L47)?